catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -0,0 +1,254 @@
+
+  public List<Suggestion> enhanceResults(List<Suggestion> results, String query) {
+    LinkedList<Suggestion> enhanced = new LinkedList<>(results);
+    LiteralSuggestion literal = getMgrsSuggestion(query);
+    if (literal != null && literal.hasGeo()) {
+      enhanced.addFirst(literal);
+    }
+    return enhanced;
+  }
+
+  @VisibleForTesting
+  List<String> getMgrsCoordinateStrings(String query) {
+    return coordinateStringsFromQuery(query);
+  }
+
+  @Nullable
+  private LiteralSuggestion getMgrsSuggestion(String query) {
+    List<String> matches = coordinateStringsFromQuery(query);
+    StringBuilder nameBuilder = new StringBuilder("MGRS: ");
+    matches.forEach(mgrs -> nameBuilder.append("[ ").append(mgrs).append(" ] "));
+    if (matches.isEmpty()) {
+      return null;
+    }
+    if (matches.size() == 1) {
+      String mgrs = matches.get(0);
+      UsngCoordinate mgrsCoord = parseMgrsString(mgrs);
+      BoundingBox bb = translator.toBoundingBox(mgrsCoord);
+      return new LiteralSuggestion("LITERAL", nameBuilder.toString(), latLonFromBoundingBox(bb));
+    }
+    List<LatLon> geo =
+        matches
+            .stream()
+            .distinct()
+            .map(this::parseMgrsString)
+            .filter(Objects::nonNull)
+            .map(translator::toLatLon)
+            .map(d -> from(d.getLat(), d.getLon()))
+            .collect(Collectors.toList());
+    return new LiteralSuggestion("LITERAL", nameBuilder.toString(), geo);
+  }
+
+  @Nullable
+  private UsngCoordinate parseMgrsString(String mgrs) {
+    try {
+      return translator.parseMgrsString(mgrs);
+    } catch (ParseException e) {
+      LOGGER.debug("Detected string was not actually valid MGRS", e);
+      return null;
+    }
+  }
+
+  private static List<LatLon> latLonFromBoundingBox(BoundingBox bb) {
+    return ImmutableList.of(
+        from(bb.getWest(), bb.getNorth()),
+        from(bb.getWest(), bb.getSouth()),
+        from(bb.getEast(), bb.getNorth()),
+        from(bb.getEast(), bb.getSouth()));
+  }
+
+  private static List<String> coordinateStringsFromQuery(String query) {
+    final Matcher matcher = PATTERN_MGRS_ANYPART.matcher(query);
+    final List<String> mgrsStrings = new ArrayList<>();
+
+    StringBuilder mgrsBuilder = new StringBuilder();
+    MgrsPartType previousType = MgrsPartType.NONE;
+
+    while (matcher.find()) {
+      String mgrsPart = matcher.group();
+      MgrsPartType type = getPartType(mgrsPart);
+      switch (type) {
+        case ZONE:
+          handleZoneWithPreviousType(mgrsPart, previousType, mgrsBuilder, mgrsStrings);
+          break;
+        case HUNDRED_KM:
+          handleHundredKmWithPreviousType(mgrsPart, previousType, mgrsBuilder);
+          break;
+        case NUMERIC:
+          handleNumericWithPreviousType(mgrsPart, previousType, mgrsBuilder, mgrsStrings);
+          break;
+        case NONE:
+          // Do nothing
```

Comment:
we should log here

Reply:
Sounds good. I ended up shifting things around. Instead of throwing the `IllegalArgumentException` in `getPartType(mgrsPart)` I just return `MgrsPartType.NONE` and throw here, at `case NONE: ...`. 

So the need to log at this location in particular is gone. Logging has also been improved through the entire class as a whole.